precompiles/erc20/query.go

```diff
@@ -51,26 +51,42 @@ func (p Precompile) Name(
 	method *abi.Method,
 	_ []interface{},
 ) ([]byte, error) {
+	// TODO: is this actually necessary here? It won't be possible to register a token
+	// with an empty denomination right? So calling this with an empty denom also won't happen..
+	if p.tokenPair.Denom == "" {
+		return nil, errors.New("denom cannot be empty")
+	}
+
```

Comment:
this shouldn't be the case, all the tokens should have it filled out

Reply:
With `not an IBC voucher` are you referring to line 67? Personally I would remove this check since it is something that should be as stated in the comment, during the registration.